scipy/signal/wavelets.py

```diff
@@ -308,14 +309,147 @@ def ricker(points, a):
     return total
 
 
-def cwt(data, wavelet, widths):
+def morlet2(M, s, w=5):
+    """
+    Complex Morlet wavelet, designed to work with `cwt`.
+
+    Returns the complete version of morlet wavelet, normalised
+    according to `s`:
+
+        ``exp(1j*w*x/s) * exp(-0.5*(x/s)**2) * pi**(-0.25) * sqrt(1/s)``
+
+    .. versionadded:: 0.18.2
+
+    Parameters
+    ----------
+    M : int
+        Length of the wavelet.
+    s : float
+        Width parameter of the wavelet.
+    w : float, optional
+        Omega0. Default is 5
+
+    Returns
+    -------
+    morlet : (M,) ndarray
+
+    See Also
+    --------
+    morlet2_freq : Conversion between wavelet width and frequency
+
+    Notes
+    -----
+    This function was designed to work with `cwt`. Because `morlet2`
+    returns an array of complex numbers, the `dtype` argument of `cwt`
+    should be set to `numpy.complex` for best results.
+
+    Note the difference in implementation with `morlet`. The fundamental
```

Comment:
It would be good to concise state in the PR summary what this difference is - the expressions seems the same. gh-3793 was about a change to ``cwt``, while this PR changes both ``cwt`` and the Morlet implementation. Maybe needed, but it's late and I can't quite formulate the rationale for it myself.

Reply:
@domengorjup A brief comment would be helpful here. There are some differences between the two implementations, which I initially assumed were because of the complex support, but the original was also supposed to support complex types.

One possible, perhaps likely, explanation is that the modifications are primarily related to the need for a change in function signature, the distinction between "complete" and "standard" versions being removed in `morlet2()`, and so on.

Reply:
The implementation differences between `morlet()` and `morlet2()` indeed have nothing to do with the complex support, but rather with the wavelet function signature requirements of `cwt()`. 

The difference in the construction of `x` simply has to do with the way the independent variable `vec` is constructed in `ricker()`, which is referenced as a sort of "template" wavelet function in the `cwt()` doc.

The wavelet function in `morlet()` is not normalized to unit norm (as is noted in its documentation), and `morlet2()` is (and should be for `cwt()` [1]). 

To the best of my knowledge (and I struggle to find clear sources here), the wavelet function, implemented in `morlet2()` is in the form, normally used in CWT.  The "complete" form in `morlet()` is also said to differ negligibly from the "standard" version when `w` is greater than 5, which is the default value of the parameter.

Continuing my answer to the above comment, the nomenclature on the Morlet and Gabor wavelets is not exactly clear to me. Perhaps `gabor` would be a more meaningful name of this wavelet function, but since it's been called the Morlet wavelet in `morlet()`, I did not want to change it.

[_[1] S. Mallat. A Wavelet Tour of Signal Processing. Elsevier Science, 1999._](https://www.di.ens.fr/~mallat/papiers/WaveletTourChap1-2-3.pdf)